File: 739-daily-temperatures/daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function(temperatures) {
  const n = temperatures.length
  const answer = new Array(n).fill(0)
  const stack = []

  for (let i = 0; i < n; i++) {
    const currentTemp = temperatures[i]

    while (stack.length > 0 && temperatures[stack[stack.length - 1]] < currentTemp) {
        let prevDay = stack.pop()
        answer[prevDay] = i - prevDay
    }
    stack.push(i)
  }

  return answer
};

File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        res = [0] * n
        stack = []

        for i, temp in enumerate(temperatures):
            while stack and temperatures[stack[-1]] < temp:
                idx = stack.pop()
                res[idx] = i - idx
            stack.append(i)

        return res